export.py

```diff
@@ -1,16 +1,74 @@
 #!/usr/bin/python3
 
-import json
 import os
-import re
+import json
+import re 
 import html2text
-import markdown
+import requests 
+from sys import exit as sysexit
 from bs4 import BeautifulSoup
+from getpass import getpass
 from datetime import datetime
+from markdown import markdown
 from operator import itemgetter
 from download_posts import download_posts
 from download_comments import download_comments
 
+# Generic headers to prevent LiveJournal from throwing out this random solicitation
+headers = {
+    "Upgrade-Insecure-Requests": "1",
+    "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/127.0.0.0 Safari/537.36 OPR/113.0.0.0",
+    "sec-ch-ua": '"Chromium";v="127"',
+    "sec-ch-ua-platform": '"Windows"',
+}
+
+try:
+    # Get a "luid" cookie so it'll accept our form login.
+    cookies = {
+        'luid': requests.get("https://www.livejournal.com/", headers=headers).headers.get('Set-Cookie').split('luid=')[1].split(';')[0]
+    }
+
+except:
+    # If attempt to reach LiveJournal fails, error out.
+    print("Couldn't reach www.livejournal.com, please check internet connection. Exiting.")
+    sysexit(1)
+
+# Populate dictionary for request
+credentials = {
+    'user': input("Enter LiveJournal Username: "),
+    'password': getpass("Enter LiveJournal Password: ")
+}
+
+# Login with user credentials and retrieve the two cookies required for the main script functions
+response = requests.post("https://www.livejournal.com/login.bml", data=credentials, cookies=cookies)
+
+# If not successful, whine about it.
+if response.status_code != 200:
+    print("Error - Return code:", response.status_code)
+
+# If successful, then get the 'Set-Cookie' key from the headers dict and parse it for the two cookies, placing them in a cookies dict
+curlOUTPUT = response.headers.get('Set-Cookie')
+ljloggedin_match = re.search(r'ljloggedin=([^\s;]+)', curlOUTPUT)
+ljmastersession_match = re.search(r'ljmastersession=([^\s;]+)', curlOUTPUT)
+cookies = {
+    'ljloggedin': ljloggedin_match.group(1) if ljloggedin_match else None,
+    'ljmastersession': ljmastersession_match.group(1) if ljmastersession_match else None,
+}
+
+# Credit to the Author!
+headers = {
+    'User-Agent': 'https://github.com/arty-name/livejournal-export; me@arty.name'
+}
+
+# If a cookie could not be successfully acquired, error out. Otherwise, get to work.
+if any(value is None for value in cookies.values()):
+    print("Could not extract cookie data despite successful login.")
```

Comment:
Similar to years handling, two similar operations could fail and we won’t know which. Having a function to get and validate value could help here, what do you think?